toolkit/scripts/check_signatures.py

```diff
@@ -20,20 +22,65 @@ def getSignature(fileName) -> str:
             sha256sum.update(read_data)
     return sha256sum.hexdigest()
 
-def check_folder(path):
+def find_file(path, filename) -> Optional[str]:
+    return_value : Optional[str] = None
+    for matching_file in Path(path).glob(f"**/{filename}"):
+        if os.path.exists(matching_file):
+            if return_value is not None:
+                print(f"ERROR: detected multiple {filename}: [{matching_files}]")
+                return None
+            
+            return_value = matching_file
+
+    return return_value
+
+def find_name_of_all_spec_and_signatures_json_pairs(path: str) -> List[str]:
+    names: List[str] = []
+    # Search for all spec files (XXX.spec)
+    for spec_path in Path(path).glob("*.spec"):
+        if os.path.exists(spec_path):
+            name = Path(spec_path).stem
+            signature_path = os.path.join(path, f"{name}.signatures.json")
+            if os.path.exists(signature_path):
+                # If there is a matching signature file (XXX.signatures.json),
+                # add it to list
+                names.append(name)
+
+    return names
+
+def find_spec_folder_ancestor(path: str) -> Optional[str]:
+    # Assume that spec/signatures.json files are only found in
+    # SPECS/XX and SPECS-EXTENDED/XX.  Find an ancestor of path
+    # that adheres to this assuption.  Return None if not found.
+    regex = f".*SPECS(-EXTENDED)?{os.sep}[^{os.sep}]+"
+    matching_path = re.search(regex, path)
+    if matching_path is not None:
```

Comment:
Not an actual change request. You mentioned you're new to Python, so just in case you find this syntax more appealing and you weren't aware of it:
```suggestion
    if matching_path:
```

This will return `False` for `None`, empty strings, empty collections (sets, lists, dictionaries), and maybe other things that I'm not aware of.